plugins/permission-node/src/ServerPermissionClient.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TokenManager, ServerTokenManager } from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import {
+  AuthorizeRequestOptions,
+  DiscoveryApi,
+  PermissionClient,
+} from '@backstage/plugin-permission-common';
+
+/**
+ * A server side {@link @backstage/plugin-permission-common#PermissionClient}
+ * that allows all backend-to-backend requests.
+ * @public
+ */
+export class ServerPermissionClient extends PermissionClient {
+  private readonly serverTokenManager: TokenManager;
+
+  constructor(options: {
+    discoveryApi: DiscoveryApi;
+    configApi: Config;
+    serverTokenManager: TokenManager;
```

Comment:
Feels oddly opinionated :p, probably make this just `tokenManager`

Reply:
The discrepancy is slightly weird at first, but I decided to intentionally leave it because for a working permission-enabled solution, you _do_ need to pass a server token manager in. I could see it either way though.